### PR TITLE
Exit with error code 1 on malformed CLI invocation [BA-3094]

### PR DESCRIPTION
**Pull request by grsterin**

### CODE REVIEW

File: src/ci/bin/testSingleWorkflowRunner.sh [hunk 1]
```diff
@@ -68,3 +68,16 @@ jq '{
...
+
+# Test 4: application should return non-zero exit code when parsing of command-line arguments fails
+set +e
+java \
+    -jar "${CROMWELL_BUILD_CROMWELL_JAR}" \
+    run ./centaur/src/main/resources/standardTestCases/hello/hello.wdl \
+    ./centaur/src/main/resources/standardTestCases/hello/hello.inputs
+retVal=$?
+if [ $retVal -eq 0 ]; then
+    echo "ERROR: application exited with exit code 0 when invalid command-line arguments were provided"
```
Comment (mcovarr) [inline src/ci/bin/testSingleWorkflowRunner.sh]: is it possible to assert that the invocation failed for invalid arguments and not some other reason?

Comment (grsterin) [reply src/ci/bin/testSingleWorkflowRunner.sh]: Yes, we can use specific exit code in case of invalid arguments.

Comment (aednichols) [reply src/ci/bin/testSingleWorkflowRunner.sh]: Could we also `tee` the standard out to a file and evaluate the file contents for specific strings? We do this earlier in this file, lines 20-24

File: src/ci/bin/testSingleWorkflowRunner.sh [hunk 2]
```diff
@@ -68,3 +68,23 @@ jq '{
...
+  exit 1
+fi
+
+# Test 4: application should return non-zero exit code when parsing of command-line arguments fails
+set +e
+java \
+    -jar "${CROMWELL_BUILD_CROMWELL_JAR}" \
+    run ./centaur/src/main/resources/standardTestCases/hello/hello.wdl \
+    ./centaur/src/main/resources/standardTestCases/hello/hello.inputs \
+2>&1 | tee console_output.txt
```
Comment (cjllanwarne) [inline src/ci/bin/testSingleWorkflowRunner.sh]: Inspired by https://stackoverflow.com/questions/692000/how-do-i-write-stderr-to-a-file-while-using-tee-with-a-pipe -

it looks like you can tee only stderr using `2> >(tee -a stderr.log >&2)`... but I'll leave it up to you whether you consider that an improvement!

File: src/ci/bin/testSingleWorkflowRunner.sh [hunk 3]
```diff
@@ -68,3 +68,23 @@ jq '{
...
+
+# Test 3: program should exit with error in case if validation of command line arguments failed
+java -jar "${CROMWELL_BUILD_CROMWELL_JAR}" run nonexistent.wdl &
+pid=$!
+sleep 10
+if kill -0 $pid > /dev/null 2>&1; then
+  echo "ERROR: Process still exists"
+  kill $pid
+  exit 1
+fi
```
Comment (cjllanwarne) [inline src/ci/bin/testSingleWorkflowRunner.sh]: Did something go awry in the rebase/merge?

File: src/ci/bin/testSingleWorkflowRunner.sh [hunk 4]
```diff
@@ -68,3 +68,23 @@ jq '{
...
+
+# Test 4: application should return non-zero exit code when parsing of command-line arguments fails
+set +e
+java \
+    -jar "${CROMWELL_BUILD_CROMWELL_JAR}" \
+    run ./centaur/src/main/resources/standardTestCases/hello/hello.wdl \
+    ./centaur/src/main/resources/standardTestCases/hello/hello.inputs \
+2>&1 | tee console_output.txt
+set -e
+grep "Error: Unknown argument" console_output.txt
```
Comment (cjllanwarne) [inline src/ci/bin/testSingleWorkflowRunner.sh]: Should probably also check that `$?` was `1` (per the PR title)

Comment (grsterin) [reply src/ci/bin/testSingleWorkflowRunner.sh]: Fixed. Lost that part, while adding error message verification.

File: src/ci/bin/testSingleWorkflowRunner.sh (hunk 3 above)
Comment (grsterin) [reply src/ci/bin/testSingleWorkflowRunner.sh]: yes, there was a conflict due to testSingleWorkflowRunner.sh being modified both in this branch and develop

**Pull request merged**